optimade/models/jsonapi.py

```diff
@@ -193,16 +200,27 @@ class ResourceLinks(BaseModel):
 class Attributes(BaseModel):
     """
     Members of the attributes object ("attributes\") represent information about the resource object in which it's defined.
-    The keys for Attributes must NOT be:
+    The keys for Attributes MUST NOT be:
         relationships
         links
         id
         type
     """
 
+    relationships: Optional[Any] = Schema(..., description="Not allowed key")
+    links: Optional[Any] = Schema(..., description="Not allowed key")
+    id: Optional[Any] = Schema(..., description="Not allowed key")
+    type: Optional[Any] = Schema(..., description="Not allowed key")
+
     class Config:
         extra = "allow"
 
+    @validator("relationships", "links", "id", "type")
+    def check_illegal_attributes_fields(cls, v):
+        raise AssertionError(
```

Comment:
When I try to add a test for this (see my commit), this gets raised as an AssertionError rather than a ValidationError (as I'd expect). Do we want to make this return a validation error instead?

Reply:
Sure. But isn't that a bit weird? Not the change, but that you're not catching it. We should have an AssertionError in the structure model as well. Do you catch that one?
Should I change it to ValueError or a ValidationError?

Reply:
I think the point is that pydantic is meant to internally catch AssertionErrors (it worked with just assert statements right?) so I'm a bit confused why this doesn't work (see https://pydantic-docs.helpmanual.io/usage/validators/). ValueError would probably be the next most appropriate?